scipy/optimize/zeros.py

```diff
@@ -352,8 +352,16 @@ class of similar problems can be solved together.
                     full_output, (p, funcalls, itr + 1, _ECONVERGED))
             p0, q0 = p1, q1
             p1 = p
-            q1 = func(p1, *args)
-            funcalls += 1
+            while True:
```

Comment:
Depending on the function, this has the potential to be an infinite loop. I would increase the iteration count, as you propose, and test if the iteration count exceeds `maxiter`.

Reply:
Ha, I typed the same thing :) so +1

Reply:
Fair enough! Done